tests/test_user_security.py

```diff
@@ -0,0 +1,200 @@
+import enum
+import unittest
+from datetime import timedelta, datetime
+from http import HTTPStatus
+
+import jwt
+from django.conf import settings
+from django.http import HttpRequest
+from sqlalchemy import Integer, Column, VARCHAR
+
+from core_lib.data_layers.data.db.sqlalchemy.base import Base
+from core_lib.data_layers.data_access.db.crud.crud import CRUD
+from core_lib.data_layers.data_access.db.crud.crud_data_access import CRUDDataAccess
+from core_lib.data_transform.result_to_dict import result_to_dict
+from core_lib.rule_validator.rule_validator import ValueRuleValidator, RuleValidator
+from core_lib.session.jwt_token_handler import JWTTokenHandler
+from core_lib.session.security_handler import SecurityHandler
+from core_lib.session.user_security import UserSecurity
+from core_lib.web_helpers.decorators import handle_exceptions
+from core_lib.web_helpers.django.decorators import RequireLogin
+from core_lib.web_helpers.request_response_helpers import response_status
+from core_lib.web_helpers.web_helprs_utils import WebHelpersUtils
+from tests.test_data.test_utils import connect_to_mem_db
+
+
+# CRUD SETUP
+class User(Base):
+    __tablename__ = 'user_security'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    username = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    role = Column(Integer, nullable=False)
+
+
+class UserDataAccess(CRUDDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('username', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('role', int),
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, User, connect_to_mem_db(), UserDataAccess.rules_validator)
+
+
+# USER SECURITY SETUP
+class PolicyRoles(enum.Enum):
+    USER = 1
+    TESTER = 2
+    ADMIN = 3
+
+
+class SessionUser(object):
+
+    def __init__(self, id: int, email: str):
+        self.id = id
+        self.email = email
+
+    def __dict__(self):
+        return {'id': self.id, 'email': self.email}
+
+
+class CustomerSecurity(UserSecurity):
+
+    def __init__(self, cookie_name: str, secret: str, expiration_time: timedelta):
+        UserSecurity.__init__(self, cookie_name, JWTTokenHandler(secret, expiration_time))
+
+    def secure_entry(self, request, session_obj: SessionUser, policies: list):
+        data_dict = result_to_dict(user_data_access.get(session_obj.id))
+        if data_dict['email'] == session_obj.email:
+            if data_dict['role'] in policies:

```

Comment:
fix policies 
create a separate function on line 64, called `def has_access(self, user_policy, check_policies)`

and add another test 

self.assertEqual(has_access(.., [] ), True)
self.assertEqual(has_access(.., [..] ), True)